test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala

```diff
@@ -945,6 +949,197 @@ class RepresenteeControllerSpec
+              .text shouldBe (messageFromMessageKey("representeeConfirmPerson.summaryline1"))
+            doc
+              .select("#name-answer")
+              .text shouldBe "First Last"
+            doc
+              .select("#account-question")
+              .text shouldBe messageFromMessageKey("representeeConfirmPerson.summaryline2")
+            doc
+              .select("#content > article > form")
+              .attr("action") shouldBe routes.RepresenteeController.confirmPersonSubmit().url
+            doc
+              .select("#confirmed > legend > h2")
+              .text() shouldBe messageFromMessageKey("representeeConfirmPerson.formTitle")
+          }
+        )
+      }
+
+      "display the page" when {
+
+        "the user has name and id " when {
+          val requiredPreviousAnswers =
+            IncompleteRepresenteeAnswers(
+              Some(IndividualName("First", "Last")),
+              Some(RepresenteeNino(NINO("AB123456C"))),
+              Some(sample[DateOfDeath]),
+              None,
+              false
+            )
+          "show the summary" in {
+            test(
+              sessionWithStartingNewDraftReturn(
+                requiredPreviousAnswers,
+                Left(PersonalRepresentative)
+              )._1,
+              routes.RepresenteeController.enterId(),
+              expectReturnToSummaryLink = false,
+              None
+            )
+          }
+        }
+      }
+
+      "redirect the page to enter-id" when {
+
+        "the user has no id " when {
+          val requiredPreviousAnswers =
+            IncompleteRepresenteeAnswers(Some(IndividualName("First", "Last")), None, None, None, false)
+          "redirect to enter id page" in {
+            inSequence {
+              mockAuthWithNoRetrievals()
+              mockGetSession(
+                sessionWithStartingNewDraftReturn(
+                  requiredPreviousAnswers,
+                  Left(PersonalRepresentative)
+                )._1
+              )
+            }
+            checkIsRedirect(performAction(), routes.RepresenteeController.checkYourAnswers())
+          }
+        }
+      }
+
+    }
+
+    "handling requests to submit the confirm person page" must {
+      def performAction(formData: Seq[(String, String)]): Future[Result] =
+        controller.confirmPersonSubmit()(FakeRequest().withFormUrlEncodedBody(formData: _*))
+
+      behave like redirectToStartBehaviour(() => performAction(Seq.empty))
+
+      behave like nonCapacitorOrPersonalRepBehaviour(() => performAction(Seq.empty))
+
+      "show a form error" when {
+
+        val session = sessionWithFillingOutReturn(
+          IncompleteRepresenteeAnswers(
+            Some(sample[IndividualName]),
+            Some(sample[RepresenteeNino]),
+            Some(sample[DateOfDeath]),
+            None,
+            false
+          ),
+          Right(Capacitor)
+        )._1
+
+        def testConfirmPersonFormError(
```

Comment:
isn't there already a `testFormError` function you can re-use in this spec?

Reply:
Took a bit of time but done.

Reply:
Done.